pbs/pbsrequest.go

```diff
@@ -30,20 +37,42 @@ type Bids struct {
 	Params     json.RawMessage `json:"params"`
 }
 
+type PBSVideo struct {
+	Mimes          []string `json:"mimes,omitempty"`
+	Minduration    uint64   `json:"minduration,omitempty"`
+	Maxduration    uint64   `json:"maxduration,omitempty"`
+	Startdelay     int64    `json:"startdelay,omitempty"`
+	Skippable      int      `json:"skippable,omitempty"`
+	PlaybackMethod int8     `json:"playback_method,omitempty"`
+	Frameworks     []string `json:"frameworks,omitempty"`
+}
+
 type AdUnit struct {
-	Code     string           `json:"code"`
-	TopFrame int8             `json:"is_top_frame"`
-	Sizes    []openrtb.Format `json:"sizes"`
-	Bids     []Bids           `json:"bids"`
-	ConfigID string           `json:"config_id"`
+	Code       string           `json:"code"`
+	TopFrame   int8             `json:"is_top_frame"`
+	Sizes      []openrtb.Format `json:"sizes"`
+	Bids       []Bids           `json:"bids"`
+	ConfigID   string           `json:"config_id"`
+	MediaTypes []string         `json:"media_types"`
 }
 
 type PBSAdUnit struct {
-	Sizes    []openrtb.Format
-	TopFrame int8
-	Code     string
-	BidID    string
-	Params   json.RawMessage
+	Sizes      []openrtb.Format
+	TopFrame   int8
+	Code       string
+	BidID      string
+	Params     json.RawMessage
+	Video      PBSVideo
+	MediaTypes []MediaType
```

Comment:
That's a very good point. Indeed, PBSAdUnit with missing Video and mediaTypes including MEDIA_TYPE_VIDEO will be invalid (unless having Video without any Video specific information is OK - I am not sure about that, would appreciate clarification here). 

However, I feel that implementing Video as a structure and Banner as an empty interface might be inconsistent and confusing, even with documentation. Besides that, I think that explicit providing of media types is a much cleaner method to control which media types are available. After all, ad unit with Video structure present but with only MEDIA_TYPE_BANNER specified is perfectly valid. 

I will fix the code so it does not fail if MEDIA_TYPE_VIDEO is specified, but the structure is absent.